registry/job.go

```diff
@@ -127,18 +130,14 @@ func (r *EtcdRegistry) Units() ([]job.Unit, error) {
 		if u == nil {
 			continue
 		}
-		uMap[u.Name] = u
-	}
-
-	var sortable sort.StringSlice
-	for name, _ := range uMap {
-		sortable = append(sortable, name)
-	}
-	sortable.Sort()
+		// Just a guard
+		_, ok := uMap[u.Name]
+		if ok {
+			continue
+		}
```

Comment:
Note that this slightly changes behaviour: previously, if there was a duplicate name, the last one would win; now it's the first...

I wonder though whether this is something that can actually happen at all? Maybe we can just drop this checking altogether.

Even if the check actually *is* necessary, we don't really need the map either: as we know the input is already sorted by name, we just need to compare each entry against the previous one.